engine/table/src/test/java/io/deephaven/engine/table/impl/TestRollup.java

```diff
@@ -132,6 +135,56 @@ private void testRollupIncrementalInternal(final String ctxt, final int size) {
         }
     }
 
+    @Test
+    public void testRollupWithFilter() {
+        final Table source = newTable(
+                stringCol("GRP", "v1", "v1", "v2", "v2", "v3"),
+                intCol("AVG", 1, 2, 3, 4, 5),
+                intCol("CONST", 11, 12, 13, 14, 15));
+        final List<Aggregation> aggs = List.of(AggAvg("avg=AVG"));
+
+        final RollupTable rollup1 = source.rollup(aggs, true, "GRP");
+        final IllegalArgumentException ex1 = Assert.assertThrows(IllegalArgumentException.class,
+                () -> rollup1.withFilter(WhereFilterFactory.getExpression("AVG >= 13")));
+        assertEquals("Invalid filter found: RangeFilter(AVG greater than or equal to 13) may only use " +
+                "non-aggregation columns, which are [GRP, CONST], but has used [AVG]", ex1.getMessage());
+        final RuntimeException ex2 = Assert.assertThrows(RuntimeException.class,
+                () -> rollup1.withFilter(WhereFilterFactory.getExpression("BOGUS = 13")));
+        assertEquals("Column \"BOGUS\" doesn't exist in this table, available columns: [GRP, AVG, CONST]",
+                ex2.getMessage());
+
+        final RollupTable rollup2 = source.rollup(aggs, true, "GRP")
+                .withFilter(WhereFilterFactory.getExpression("GRP = `v2`"));
+        final Table snapshot2 = snapshotFilteredRollup(rollup2);
```

Comment:
We can now tell how many times the GRP column was read, in the first case it should be zero.  In the other two cases it should be non-zero (resetting between each).
```suggestion
        grpCount.set(0);
        final RollupTable rollup2 = rollup1
                .withFilter(WhereFilterFactory.getExpression("GRP = `v2`"));
        assertEquals(0, grpCount.get());
        final Table snapshot2 = snapshotFilteredRollup(rollup2);

```